lv2/thread/thread.ttl

```diff
@@ -0,0 +1,54 @@
+@prefix doap:   <http://usefulinc.com/ns/doap#> .
+@prefix foaf:   <http://xmlns.com/foaf/0.1/> .
+@prefix lv2:    <http://lv2plug.in/ns/lv2core#> .
+@prefix opts:   <http://lv2plug.in/ns/ext/options#> .
+@prefix owl:    <http://www.w3.org/2002/07/owl#> .
+@prefix rdf:    <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs:   <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix thread: <http://lv2plug.in/ns/ext/thread#> .
+@prefix xsd:    <http://www.w3.org/2001/XMLSchema#> .
+
+<http://lv2plug.in/ns/ext/thread>
+	a owl:Ontology ;
+	rdfs:seeAlso <thread.h> ,
+		<lv2-thread.doap.ttl> ;
+	lv2:documentation """
+<p>This extension defines two instantiation options: thread:schedPolicy and
+thread:schedPriority. Plugins that require custom realtime background
+threads should call <code>pthread_setschedparam()</code> with the given
+parameters.</p>
+
+<p>This allows which require background threads to set-up correct realtime
+schedule priorities for those threads, that will not interfere with host
+scheduling.<p>
+
+<p>Hosts should pass values appropriate to not interfere with their
+internal processing. e.g. jack hosts should use
+<code>jack_client_real_time_priority</code> when JACK runs with
+realtime scheduling.<p>
+
+<p>This extension is intended to be used with pthread on Unix systems
+and potentially pthread-w32 on Windows. MacOS plugins that use mach-
+threads can combine <code>pthread_mach_thread_np</code> with time-
+constraints passed via the bufsiz extension when calling
+<code>thread_policy_set</code. The sched policy is applicable as-is
+on all OS.</p>
+""" .
+
+thread:schedPolicy
+	a rdf:Property ,
+		opts:Option ,
+		rdfs:range xsd:nonNegativeInteger;
```

Comment:
As annoying as it is to specify this stuff, I'm really unsure about "hey, y'know, some number that means a policy of some sort"...

I'm wondering if we actually need this one?  Priority is easy, it's a number with pretty clear semantics (maybe the range is a bit weird and platform specific, but, meh, close enough).

Reply:
pthread_setschedparam(.., int policy, ..);
an `int` is used by pthread to abstract this, depending on OS.

Reply:
pthread is not the only threading API in the world.  That approach is only really used by projects that take the "just export all my Unix stuff to these other platforms" approach to portability, i.e. grudging ports from Linux land.  I don't think it's a good thing for the official specs to have that particular smell.